src/main/java/recipe/RecipePresenter.java

```diff
@@ -1,34 +1,33 @@
 package recipe;
 
-import ui.CreateRecipeUI;
-import ui.ViewRecipeUI;
-
 public class RecipePresenter implements RecipeOutputBoundary{
-    CreateRecipeUI createRecipeUI;
-    ViewRecipeUI viewRecipeUI;
+    RecipeCreateBox recipeCreateBox;
+    RecipeViewBox recipeViewBox;
 
-    public RecipePresenter(CreateRecipeUI createRecipeUI) {
-        this.createRecipeUI = createRecipeUI;
+    @Override
+    public void createSuccessView() {
+        recipeCreateBox.success();
     }
 
-    public RecipePresenter(ViewRecipeUI viewRecipeUI){
-        this.viewRecipeUI = viewRecipeUI;
+    @Override
+    public void createFailureView() {
+        recipeCreateBox.fail();
     }
 
-
     @Override
-    public void createSuccessView() {
-        createRecipeUI.success();
+    public void readSuccessView(String result) {
+        recipeViewBox.success(result);
     }
 
     @Override
-    public void createFailureView() {createRecipeUI.failure();}
+    public void readFailureView(){
+        recipeViewBox.failure();
```

Comment:
A bit confused based on just the names what the recipeCreateBox.fail and recipeViewBox.failure methods are for. Maybe some javadoc documentation can help?